### DISCUSSION TITLE
如何从源代码编译安装1panel？

### DISCUSSION THREAD

**Discussion by panluoyant** [1Panel-dev/1Panel#5746] [Q&A]
如题

**Comment by wanghe-fit2cloud** [comment]
> 如题

官方文档中有相关描述。

**Reply by panluoyant** [reply]
我刚刚查看了官方文档，文档中仅提到在线使用脚本安装，以及在离线环境下使用安装包进行安装，没有提到使用源代码进行编译安装

**Reply by wanghe-fit2cloud** [reply]
<img width="277" alt="image" src="https://github.com/1Panel-dev/1Panel/assets/46734380/39a4258b-32ce-46a9-b3ae-fc7ae28941e7">

**Reply by panluoyant** [reply]
谢谢，我刚刚可能是没看到，因为我是想直接在服务器上进行构建安装，所以我并没有看开发的那个文档，非常感谢

**Reply by panluoyant** [reply]
![image](https://github.com/user-attachments/assets/eb6de867-e9ed-427e-a5d1-fb79b6e97bbc)
请问这个咋解决，这个base-dir变量应该如何设置？

**Reply by zhengkunwang223** [reply]
> ![image](https://private-user-images.githubusercontent.com/104615547/348551733-eb6de867-e9ed-427e-a5d1-fb79b6e97bbc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjEwOTcyNTAsIm5iZiI6MTcyMTA5Njk1MCwicGF0aCI6Ii8xMDQ2MTU1NDcvMzQ4NTUxNzMzLWViNmRlODY3LWU5ZWQtNDI3ZS1hNWQxLWZiNzliNmU5N2JiYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcxNlQwMjI5MTBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jZDU4ZjFmZTdkMDIzZjUyOThkNjMyYzIwNzhmZGE5MGFmNjZmYTdjNDg5YzQ5Y2UzNjc4Y2QyNzU4YjUwZTY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.KEcNwAOuYJtk3_MTPG7PqXt7neueW6Bey6ZS9U13eUg) 请问这个咋解决，这个base-dir变量应该如何设置？

代码相关的问题 需要你自行处理
